src/WarcraftLegacies.Source/Mechanics/Scourge/HelmOfDominationDropsWhenScourgeLeaves.cs

```diff
@@ -1,22 +1,25 @@
-// using WarcraftLegacies.Source.Legends;
-// using WarcraftLegacies.Source.Setup;
-// using WarcraftLegacies.Source.Setup.FactionSetup;
-//
-// using static War3Api.Common;  using static MacroTools.GeneralHelpers;
-// {
-//   public class HelmOfDominationDropsWhenScourgeLeaves{
-//
-//     private static void Actions( ){
-//       if (GetTriggerFaction() == ScourgeSetup.FactionScourge && ArtifactSetup.ArtifactHelmofdomination.OwningUnit == LegendScourge.LegendLichking.Unit){
-//         SetItemPosition(ArtifactSetup.ArtifactHelmofdomination.Item, GetRectCenterX(gg_rct_LichKing), GetRectCenterY(gg_rct_LichKing));
-//       }
-//     }
-//
-//     public static void Setup( ){
-//       trigger trig = CreateTrigger();
-//       OnFactionGameLeave.register(trig);
-//       TriggerAddAction(trig,  Actions);
-//     }
-//
-//   }
-// }
+using WarcraftLegacies.Source.Setup.Legends;
+using WarcraftLegacies.Source.Setup;
+using WarcraftLegacies.Source.Setup.FactionSetup;
+using static War3Api.Common;
+using MacroTools.FactionSystem;
+using MacroTools.Extensions;
+
+public class HelmOfDominationDropsWhenScourgeLeaves
+{
+    public static void Setup()
+    {
+        Faction.GameLeave += Faction_GameLeave;
+    }
+
+    private static void Faction_GameLeave(object? sender, Faction e)
+    {
+        if (e == ScourgeSetup.Scourge && ArtifactSetup.ArtifactHelmofdomination.OwningUnit == LegendScourge.LegendLichking.Unit)
+        {
+            var lkPos = LegendScourge.LegendLichking.Unit.GetPosition();
+            LegendScourge.LegendLichking.Unit.DropAllItems();
+            SetItemPosition(ArtifactSetup.ArtifactHelmofdomination.Item, lkPos.X - 55, lkPos.Y + 30); // Move item in front of the LK 
```

Comment:
```suggestion
            ArtifactSetup.ArtifactHelmofdomination?.Item.SetPosition(new Point(lichKingPosition.X - 55, lichKingPosition.Y + 30));
```
Something like this will stop the compiler complaining about the possible null reference, and it's a little more standard (use Points where possible). But it is of course a little uglier.